### PR TITLE
Makings BigtableOptionsFactory usable without an HBase Configuration 

### PR DESCRIPTION
**Pull request by sduskis**
Moving BigtableOptionsFactory to bigtable-grpc so that the connection configuration can be done without having an HBase Configuration object.

This is a step towards getting a Bigtable API that's independent of hbase

### CODE REVIEW

File: bigtable-hbase/src/test/java/com/google/cloud/bigtable/hbase/TestClusterAPI.java [hunk 1]
```diff
@@ -67,6 +67,46 @@
...
+
+    @Override
+    public String get(String key, String defaultValue) {
+      return System.getProperty(key, defaultValue);
+    }
+
+    @Override
+    public int getInt(String key, int defaultValue) {
+      if (System.getProperties().contains(key)) {
+        return new Integer(get(key));
```
Comment (kevinsi4508) [inline bigtable-hbase/src/test/java/com/google/cloud/bigtable/hbase/TestClusterAPI.java]: Integer.parseInt()?

**Comment by kevinsi4508** [comment]
LGTM

**Comment by AngusDavis** [comment]
While I don't have any particular issues with this PR - I don't think we'll want to have the BigtableOptionsFactory used outside of Hadoop. The key names are all very verbose and Hadoop-y. I'd expect non-HBase clients to instead make use of a BigtableOptions Builder of some sort.

**Comment by sduskis** [comment]
@AngusDavis: I agree with you with the details.  I guess what I'm really aiming for is a way to set defaults for BigtableOptions.  Right now, the hadoopy names and default values are tightly coupled.  Let me see if I can separate those out.

**Pull request closed without merging**